lib/migration.pm

```diff
@@ -157,4 +158,26 @@ sub reset_consoles_tty {
     reset_consoles;
 }
 
+# Assert read-only snapshot before migrated
+# Assert screen 'linux-login' with 200s
+# Workaround known issue: bsc#980337
+# In this case try to select tty1 with multi-times then select root console
+sub boot_into_ro_snapshot {
+    if (!check_screen('linux-login', 200)) {
+        record_soft_failure 'bsc#980337';
+        my $retries = 10;
+        for my $i (0 .. $retries) {
+            send_key "ctrl-alt-f1";
+            if (check_screen('tty1-selected', 5)) {
+                last;
+            }
+            else {
+                record_info('not in tty1', 'switch to tty1 failed', result => 'softfail');
+            }
+            die "Boot into read-only snapshot failed over 5 minutes, considering a product issue" if $i == $retries;
+            sleep 10;
```

Comment:
You could also put this in checkscreen, and have timeout there, so we have chance to match it earlier.

Reply:
use checkscreen here is no sense, tty switch by sending key.

Reply:
Sorry for the confusion, that's not what I mean, there is already check-screen above, instead of sleep we could wait longer to match tty1 instead of idle time. 
Imagine following, tty1 appear in 5 seconds, with your current implementation we will wait 12 seconds before we match it, if we put 10 seconds, we will match it immediately.
In case it doesn't appear we will still wait 10 seconds before retry. Hope the point is clear now.

Reply:
yes, hate checkscreen :)